### PR TITLE
Feature/add moonjs

### PR DESCRIPTION
**Pull request by juanpflores**
Pull request for issue: #11291 
Related issue(s): NA

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/KingPixil/moon
 * Official website (optional, not the repository): http://moonjs.ga/
 * NPM package url (optional): https://www.npmjs.com/package/moonjs
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 32
   - Count of stars: 576
   - Count of forks: 22
 * NPM download stats (optional):
   - Downloads in the last day: 55
   - Downloads in the last week: 440
   - Downloads in the last month: 773

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [ ] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [ ] The parent of the commit(s) in the PR is not old than 3 days.
 * [ ] Pull request is sending from a non-master branch with meaningful name.
 * [ ] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/moonjs/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "name": "moonjs",
+  "filename": "moon.min.js",
+  "version": "0.10.0",
+  "description": "A minimal, blazing fast UI library.",
+  "homepage": "http://moonjs.ga/",
+  "keywords": [
+    "UI",
+    "MVC",
+    "Library",
+    "JS"
```
Comment (pvnr0082t) [inline ajax/libs/moonjs/package.json]: We can use the keywords from upstream. Ref: https://github.com/KingPixil/moon/blob/v0.10.0/package.json#L16-L19

File: ajax/libs/moonjs/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "type": "git",
+    "url": "https://github.com/KingPixil/moon"
+  },
+  "license": "MIT",
+  "npmName": "moonjs",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "moon*.js"
```
Comment (pvnr0082t) [inline ajax/libs/moonjs/package.json]: Let's use `"**/*"` instead.

**Comment by maruilian11** [comment]
@juanpflores Do you meet any problem we can help you?

**Comment by juanpflores** [comment]
@maurilian11 I'm working on making the changes cleaner in another branch. What I wanted to do is 

1. Push Everything with one commit.
1. Avoid downloading all the repo to push one library or using the Github web client to add files. 
1. I'm having some troubles with md5 `validation` which si basically the source of all the problems above.

**Comment by maruilian11** [comment]
@juanpflores hmmm.. not sure about about the md5 validation solution. 
I will recommend you to use GitHub web UI to send the PR, we will help you squash the commits, you   can modify lots of time until the the reviewer accept it. And we will make them one clean and clear commit. Feel free to ask if you meet problems. 👍

**Pull request closed without merging**